src/app/plugins/tictactoe.plugin.ts

```diff
@@ -0,0 +1,255 @@
+  };
+
+  private _choosing: Choosing = Choosing.Row;
+  private _winner: Maybe<number> = null;
+  private _isTie: boolean = false;
+  private _row = -1;
+  private _col = -1;
+
+  //-1 is playerA
+  private currentPlayer: number = -1;
+
+  constructor(playerA: User, playerB: User) {
+    this._playerA = playerA;
+    this._playerB = playerB;
+
+    //Make 3x3 board of 0
+    this._board = [
+      [0, 0, 0],
+      [0, 0, 0],
+      [0, 0, 0],
+    ];
+  }
+
+  showBoard() {
+    const boardAsString = this._board
+      //Convert each element of each row into an emoji
+      //Join each column with a space, each row with a newline
+      .map((row) => row.map((col) => this._flagToEmoji[col]).join(' '))
+      .join('\n');
+
+    const embed = new MessageEmbed();
+    embed.setTitle('Tic Tac Toe');
+    embed.setDescription(boardAsString);
+
+    if (this._winner) {
+      embed.setDescription(
+        `${boardAsString}\n**` +
+          `${this._winner === -1 ? this._playerA.username : this._playerB.username}` +
+          `** is the winner!`
+      );
+      return embed;
+    }
+
+    if (this._isTie) {
+      embed.setDescription(`${boardAsString}\n**It's a tie!**`);
+      return embed;
+    }
+
+    const bold = (s: string) => {
+      return `**${s}**`;
+    };
+
+    const playerATitle =
+      this.currentPlayer === -1 ? bold(this._playerA.username) : this._playerA.username;
+
+    const playerBTitle =
+      this.currentPlayer === 1 ? bold(this._playerB.username) : this._playerB.username;
+
+    embed.setDescription(`${boardAsString}\n${playerATitle} vs ${playerBTitle}`);
+    return embed;
+  }
+
+  public getCurrentPlayer() {
+    if (this.currentPlayer === -1) {
+      return this._playerA;
+    }
+    return this._playerB;
+  }
+
+  public reset() {
+    this._choosing = Choosing.Row;
+    this._row = -1;
+    this._col = -1;
+  }
+
+  public async choose(index: number, msg: IMessage) {
+    if (this._choosing === Choosing.Row) {
+      this._row = index;
+      this._choosing = Choosing.Row;
+      return;
+    }
+
+    this._col = index;
+
+    //Make the move -------------------
+
+    //Make sure its not overwriting
+    if (this._board[this._col][this._row] !== 0) {
+      this.reset();
+      return;
+    }
+
+    this._board[this._col][this._row] = this.currentPlayer;
+    if (this._checkWin()) {
+      this._winner = this.currentPlayer;
+    }
+
+    this._checkTie();
+
+    this.currentPlayer *= -1;
+    await msg.edit(this.showBoard());
+    //Reset where the player is choosing
+    this.reset();
+  }
+
+  private flipTurn() {
+    this.currentPlayer *= -1;
+  }
+
+  private _checkWin() {
+    let flag = false;
+    this._board.forEach((row) => {
+      //If the sum of the row is 3, someone has won
+      if (Math.abs(this._sumArray(row)) === 3) {
+        flag = true;
+      }
+    });
+
+    for (let i = 0; i < 3; i++) {
+      //Get the column
+      const col = [];
+      for (let j = 0; j < 3; j++) {
```

Comment:
I don't like these kinds of for loops, but don't know a better way to do this
Lmk if someone knows a better way to get the columns

Reply:
there may be an array transpose func